src/error.rs

```diff
@@ -66,28 +68,16 @@ where
 impl<T, I> error::Error for Error<T, I>
 where
     T: Sink<I> + TryStream,
-    <T as Sink<I>>::Error: error::Error,
-    <T as TryStream>::Error: error::Error,
+    <T as Sink<I>>::Error: error::Error + 'static,
```

Comment:
Ah, this is a backwards-incompatible change, which means we'll need to bump the major version. Just an OBS to whomever ends up pushing this change out.